app/controllers/posts/likes_controller.rb

```diff
@@ -0,0 +1,12 @@
+class Posts::LikesController < ApplicationController
+  def create
+    @like_post = current_user.likes.create!(post_id: params[:post_id])
+    redirect_to request.referer || root_path
+  end
+
+  def destroy
+    @like_post = current_user.likes.find_by(post_id: params[:post_id])
```

Comment:
`@like_post` がnilになったらdestroy!で落ちるので`!`つける
```suggestion
    @like_post = current_user.likes.find_by!(post_id: params[:post_id])
```